src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs

```diff
@@ -408,17 +408,17 @@ static void AppendTypeName(Utf8StringBuilder sb, MetadataType t)
                     }
                     else if (type is IPrefixMangledMethod)
                     {
-                        mangledName = GetPrefixMangledMethodName((IPrefixMangledMethod)type).ToString();
+                        mangledName = new Utf8String(GetPrefixMangledMethodName((IPrefixMangledMethod)type).ToString());
                     }
                     else if (type is IPrefixMangledType)
                     {
-                        mangledName = GetPrefixMangledTypeName((IPrefixMangledType)type).ToString();
+                        mangledName = new Utf8String(GetPrefixMangledTypeName((IPrefixMangledType)type).ToString());
```

Comment:
`GetPrefixMangledTypeName` returns Utf8String. This roundtrip via String is unnecessary